spartan/src/node/replication/event.rs

```diff
@@ -0,0 +1,68 @@
+use crate::node::DB;
+use maybe_owned::MaybeOwned;
+use serde::{Deserialize, Serialize};
+use spartan_lib::core::{
+    dispatcher::{simple::PositionBasedDelete, SimpleDispatcher, StatusAwareDispatcher},
+    message::Message,
+    payload::Identifiable,
+};
+
+#[derive(Serialize, Deserialize)]
+pub enum Event {
+    Push(Message),
+    Pop,
+    Requeue(<Message as Identifiable>::Id),
+    Delete(<Message as Identifiable>::Id),
+    Gc,
+    Clear,
+}
+
+impl Event {
```

Comment:
This impl can be replaced with separate trait, and it's implementation for manager